wdl-grammar/tests/parsing/metadata/source.wdl

```diff
@@ -0,0 +1,139 @@
+# This is a test of parsing task metadata sections.
+
+version 1.1
+
+task test {
+    meta {
+        a: "hello"
+        b: 'world'
+        c: 5
+        d: -0xf
+        e: 1.0e10
+        f: -2.
+        g: true
+        h: false
+        i: null
+        j: {
+            a: [1, 2, 3],
+            b: ["hello", "world", "!"],
```

Comment:
I believe metadata arrays must all be of the same type, but I don't see that tested/checked. What happens when the WDL looks like:
```suggestion
            a: [1, 2, 3, 4.5],
            b: ["hello", "world", "!", false],
```

Reply:
You're right! However, this is just a syntax tree, which would accept arrays of any metadata value type regardless of the actual expressed type as that's what the grammar allows.

Type checking of metadata values can be done statically because they don't accept expressions and may be implemented during a future validation pass of the CST.

Type checking for (the soon-to-be PR'd) expressions would require a proper resolution of one or more ASTs (if there are `import` statements); we'll probably implement that in the future as part of a real execution engine.

Reply:
That said, I've been trying to stick to as-syntactically-valid as possible tests so that a future validation pass doesn't require an update to these baselines (said validation would then have test casts for mismatched types in arrays).

Reply:
All that makes sense! Thanks for the explanation.